pkg/installmanager/installmanager.go

```diff
@@ -969,60 +949,93 @@ func writeModifiedBytesToFile(modifiedManifestBytes []byte, d fs.DirEntry, path
 	manifestBytesJson, err := yaml.YAMLToJSON(manifestBytes)
 	if err != nil {
 		return nil, errors.Wrap(err, "Error converting yaml to json")
 	}
-	/*
-		c, err := yamlutils.Decode(manifestBytes)
-		// Decoding the yaml here shouldn't produce an error. An error indicates that the
-		// installer produced yaml that could not be decoded.
-		if err != nil {
-			logger.WithError(err).Error("unable to decode manifest bytes")
-			return nil, err
-		}
-
-		// Return if file is not a MachineSet manifest
-		if isMachineSet, _ := yamlutils.Test(c, "/kind", "MachineSet"); !isMachineSet {
-			return nil, nil
-		}
-	*/
 
 	// Return if file is not a MachineSet manifest
 	if kind := gjson.Get(string(manifestBytesJson), `kind`).String(); kind != "MachineSet" {
 		return nil, nil
 	}
+	modified := false
 
-	// Match MachineSet to MachinePool name
-	msetType := gjson.Get(string(manifestBytesJson), `spec.template.metadata.labels.machine\.openshift\.io\/cluster-api-machine-type`).String()
-	if msetType == "" {
-		logger.Error("MachineSet does not contain machine.openshift.io/cluster-api-machine-type label")
-		return nil, errors.New("MachineSet does not contain machine.openshift.io/cluster-api-machine-type label")
-	}
+	if addSecurityGroup {
+		pool := poolsByType["worker"]
 
-	if pool, exists := poolsByType[msetType]; exists {
-		poolName := pool.Spec.Name
-		msetName := gjson.Get(string(manifestBytesJson), `spec.template.metadata.labels.machine\.openshift\.io\/cluster-api-machineset`).String()
-		if msetName == "" {
-			logger.WithError(err).Error("MachineSet does not contain machine.openshift.io/cluster-api-machineset label")
-			return nil, errors.New("MachineSet does not contain machine.openshift.io/cluster-api-machineset label")
-		}
-		logger.Infof("Matching machineset %s of type %s with pool %s", msetName, msetType, poolName)
+		isWorker := gjson.Get(string(manifestBytesJson), `spec.template.metadata.labels.machine\.openshift\.io\/cluster-api-machine-type`).String()
 
-		manifestBytesJson, err = sjson.SetBytes(manifestBytesJson, `metadata.labels.hive\.openshift\.io/managed`, "true")
-		if err != nil {
-			logger.WithError(err).Error("error applying hive managed label patch")
-			return nil, err
+		if isWorker == "worker" {
+			modified = true
+			var vpcIDFilterValue map[string]interface{} = map[string]interface{}{
+				"name":   "vpc-id",
+				"values": []string{vpcID},
+			}
+
+			// Add the security group name obtained from the ExtraWorkerSecurityGroupAnnotation
+			// annotation found on the MachinePool to the existing tag:Name filter in the worker
+			// MachineSet manifest
+			securityGroupFilterValue := pool.Annotations[constants.ExtraWorkerSecurityGroupAnnotation]
+
+			manifestBytesJson, err = sjson.SetBytes(manifestBytesJson, `spec.template.spec.providerSpec.value.securityGroups.0.filters.0.values.0`, securityGroupFilterValue)
```

Comment:
Eh, I think this may be wrong: it is _replacing_ the existing value, whereas the original code is _adding_.

We start with
```yaml
          securityGroups:
          - filters:
            - name: tag:Name
              values:
              - test-czcpt-worker-sg
```

and we want this line to result in
```yaml
          securityGroups:
          - filters:
            - name: tag:Name
              values:
              - test-czcpt-worker-sg
              - value-from-annotation
```

but I believe it's instead ending up as
```yaml
          securityGroups:
          - filters:
            - name: tag:Name
              values:
              - value-from-annotation
```

Need to confirm via UT. Existing tests are not sufficient: [these additions](https://github.com/openshift/hive/pull/2251/commits/6de60ba5d9af04001021d516145f63268adadebb#diff-e9eea65e1b515787afedc1c1e6f448892582a3f8604589b56e4773c3d8880a4f) (which are part of the needed rebase) get partway there, but per [this comment](https://github.com/openshift/hive/pull/2251/commits/6de60ba5d9af04001021d516145f63268adadebb#r1588428431) we still need to add a check for the above.